scanpipe/pipes/d2d.py

```diff
@@ -264,46 +265,46 @@ def get_diff_ratio(to_resource, from_resource):
     return matcher.quick_ratio()
 
 
-def _resource_path_map(to_resource, from_resources, diff_ratio_threshold=0.7):
-    path_parts = Path(to_resource.path.lstrip("/")).parts
-    path_parts_len = len(path_parts)
+def _resource_path_map(
+    to_resource, from_resources, from_resources_index, diff_ratio_threshold=0.7
+):
+    matches_iterator = pathmap.find_paths(to_resource.path, from_resources_index)
+
+    matches = list(matches_iterator)
+    if not matches:
+        return
 
-    for path_parts_index in range(1, path_parts_len):
-        current_parts = path_parts[path_parts_index:]
-        current_path = "/".join(current_parts)
+    matched_path_length, resource_ids = matches[0]
```

Comment:
Are you sure we only ever get a single match? I am not sure that there is any guarantee for this.